lib/compiler/src/v3_core.erl

```diff
@@ -2117,7 +2120,9 @@ pat_bin(Ps0, St) ->
     pat_segments(Ps, St).
 
 pat_bin_expand_strings(Es0) ->
-    foldr(fun ({bin_element,Line,{string,_,S},Sz,Ts}, Es1) ->
+    foldr(fun ({bin_element,Line,{string,_,S},default,_Ts}, Es1) ->
```

Comment:
```suggestion
    foldr(fun ({bin_element,Line,{string,_,S},default,default}, Es1) ->
```

Hi @bjorng, I forgot to run scripts/diffable. I did it just now. I think this change is necessary otherwise we will wrongly compile any encoded string.

Although I am thinking maybe we should just convert other types into integers right here... I already do a similar optimization in `v3_kernel` (but only for UTF-8).